test/VS.Web.CG.Mvc.Test/CdnScriptTagTests.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Security.Cryptography;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.VisualStudio.Web.CodeGenerators.Mvc
+{
+    public class CdnScriptTagTests
+    {
+        private readonly ITestOutputHelper _output;
+
+        public CdnScriptTagTests(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        [Fact]
+        public async Task ScriptTags_SubresourceIntegrityCheck()
+        {
+            var slnDir = GetSolutionDir();
+            var sourceDir = Path.Combine(slnDir, "src", "VS.Web.CG.Mvc");
+            var cshtmlFiles = Directory.GetFiles(sourceDir, "*.cshtml", SearchOption.AllDirectories);
+
+            var scriptTags = new List<ScriptTag>();
+            foreach (var cshtmlFile in cshtmlFiles)
+            {
+                scriptTags.AddRange(GetScriptTags(cshtmlFile));
+            }
+
+            Assert.NotEmpty(scriptTags);
+
+            var shasum = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            using (var client = new HttpClient())
+            {
+                foreach (var script in scriptTags)
+                {
+                    if (shasum.ContainsKey(script.Src))
+                    {
+                        continue;
+                    }
+
+                    using (var resp = await client.GetStreamAsync(script.Src))
+                    using (var alg = SHA384.Create())
+                    {
+                        var hash = alg.ComputeHash(resp);
+                        shasum.Add(script.Src, "sha384-" + Convert.ToBase64String(hash));
+                    }
+                }
+            }
+
+            Assert.All(scriptTags, t =>
+            {
+                Assert.True(shasum[t.Src] == t.Integrity, userMessage: $"Expected integrity on script tag to be {shasum[t.Src]} but it was {t.Integrity}: {t.Path}");
+            });
+        }
+
+        private struct ScriptTag
+        {
+            public string Src;
+            public string Integrity;
+            public string FileName;
+            internal string Path;
+        }
+
+        private static readonly Regex _scriptRegex = new Regex(@"<script[^>]*src=""(?'src'http[^""]+)""[^>]*integrity=""(?'integrity'[^""]+)""([^>]*)>", RegexOptions.Multiline);
```

Comment:
Nits:
The final capture group (unnamed) is unnecessary and can be removed, changing
`([^>]*)` to `[^>]*`

Also, performance will be better using lazy captures, changing each `*` to `*?`

An unformatted regex is difficult to read. Using RegexOptions.IgnorePatternWhitespace it can be formatted to be more readable. Including the above changes, something like:
```
        private static readonly Regex _scriptRegex = new Regex(
            @"<script[^>]*?
                src=""(?'src'http[^""]+)""[^>]*?
                integrity=""(?'integrity'[^""]+)""
                [^>]*?>"
                , RegexOptions.Multiline | RegexOptions.IgnorePatternWhitespace);
```

Reply:
Regexes are hard to understand. We should use AngleSharp in future.

Reply:
Thanks for the feedback. Feel free to change it in dev. The release/2.1 ship already sailed though.